main.js

```diff
@@ -1 +1,36 @@
 'use strict';
+let sorting = document.getElementById('sorting');
+
+sorting.onclick = function(e) {
+  if (e.target.tagName != 'TH') return;
+  sortGrid(e.target.cellIndex, e.target.getAttribute('data-type'));
+};
+
+function sortGrid(colNum, type) {
+  let tbody = sorting.getElementsByTagName('tbody')[0];
+  let rowsArray = [].slice.call(tbody.rows);
+  let compare;
+
+  switch (type) {
```

Comment:
it is ok to use switch here but function to compare need to be outside e sort function because with this solution you create a compare function on each click

Reply:
If you do not know how to do it, try to google "higher order function"